cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogDataOffsetCallback.java

```diff
@@ -44,7 +44,7 @@ public Object encodeSend(LogEvent logEvent) {
     String className = "";
     String simpleClassName = "";
     int lineNumber = 0;
-    if (stackTraceElements.length > 0) {
+    if (stackTraceElements != null && stackTraceElements.length > 0) {
```

Comment:
event.hasCallerData will return null if event.getCallerData has not been called. You can check the logic in ch.qos.logback.classic.spi.LoggingEvent.